### DISCUSSION TITLE
Android Appium Settings .apk

### DISCUSSION THREAD

**Discussion by cosminstirbu** [appium/appium#20638] [General]
Hello,

We're using Appium to test a device that runs in kiosk mode.

It seems that due to the fact the device is running in kiosk mode, Appium Settings .apk is not allow to start.

Based on its [functionality](https://github.com/appium/io.appium.settings) I'm not sure that we even need Appium Settings .apk. Can we opt out of and skip its installation and launch via a driver capability?

Basically when Appium server runs this command:

`[a5852727][ADB] Running 'adb.exe -P 5037 -s ... shell am start -n io.appium.settings/.Settings -a android.intent.action.MAIN -c android.intent.category.LAUNCHER'`

this is the error that we're seeing in logcat

`2024-10-03 08:44:48.576   597-916   ActivityTaskManager     system_server                        E  Attempted Lock Task Mode violation r=ActivityRecord{afda57f u0 io.appium.settings/.Settings}`

**Comment by KazuCocoa** [comment]
Have you tried `appium:skipDeviceInitialization`?
https://github.com/appium/appium-uiautomator2-driver

**Reply by cosminstirbu** [reply]
It works when using `appium:skipDeviceInitialization`.

Can you please expand on what are the implications of using it? In the docs it says the following:

_If set to `true` then device startup checks (whether it is ready and whether Settings app is installed) will be canceled on session creation. Could speed up the session creation if you know what you are doing. `false` by default_

Can you also tell me in what repository is this flag used, so I can have a closer look at the source code to understand its implications?

**Reply by KazuCocoa** [reply]
https://github.com/appium/appium-android-driver/blob/515423dbbd4ee6867e157f97a427e82ddab8195f/lib/commands/device/common.js#L206-L229

**Reply by cosminstirbu** [reply]
Thank you!
